flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -989,19 +1006,6 @@ private void releaseOutputResources() throws Exception {
         }
     }
 
-    private Exception runAndSuppressThrowable(
-            ThrowingRunnable<?> runnable, @Nullable Exception originalException) {
-        try {
-            runnable.run();
-        } catch (Throwable t) {
-            // TODO: investigate why Throwable instead of Exception is used here.
-            Exception e = t instanceof Exception ? (Exception) t : new Exception(t);
```

Comment:
If I understand the new code, you changed a bit that behaviour. We no longer close other resources if a Throwable that does not extend from an exception is thrown, right?

Reply:
Yes, it is true. But according to `TODO` above it wasn't clear why we used `Throwable` at all. I personally think that it is not a problem that in case of unexpected throwable we lost other exceptions because this throwable will be the much more important. But of course, we can discuss it

Reply:
Honestly, I don't feel confident making that decision. It's quite an outer layer of the entire Flink stack, so actually it might make sense to really catch all sorts of errors and try to clean up the resources.

@pnowojski WDYT?

Reply:
One more point to the discussion, which actually makes me prefer keeping the old behaviour.

In `TaskCanceler` we catch `Throwable` and rethrow it only if it is one of `InternalError`, `UnknownError`, `ThreadDeath` all other are ignored. If we ignore an error we should have a proper resource cleaning for such an error and thus we should not skip resource cleaning for `Throwables`, imo.

Reply:
Ok, I have left the old behavior. Please take a look at the last commit.